localization/localization_util/src/util_func.cpp

```diff
@@ -18,6 +18,32 @@
 
 static std::random_device seed_gen;
 
+std::vector<Eigen::Vector2d> create_initial_pose_offset_model(
+  const std::vector<double> & x, const std::vector<double> & y)
+{
+  int size = x.size();
+  std::vector<Eigen::Vector2d> initial_pose_offset_model(size);
+  for (int i = 0; i < size; i++) {
+    initial_pose_offset_model[i].x() = x[i];
+    initial_pose_offset_model[i].y() = y[i];
+  }
+
+  return initial_pose_offset_model;
+}
+
+Eigen::Matrix2d calc_eigenvector_angle(const Eigen::Matrix2d & matrix)
```

Comment:
This function returns `Eigen::Matrix2d::Zero()` when `eigensolver.info()` is not Success, which means failure.
I would like to see you use `std::optional` or `exception` so that failure can be more explicitly identified.
If failures occur with a not-insignificant probability, I would suggest using `std::optional`, and if failures rarely occur, I would suggest using `exception`.

Reply:
Also, please change the function name like followings :pray:
* `compute_rotation_matrix_to_decompose_by_pca(const Eigen::Matrix2d & covariance_matrix)`
* `find_rotation_matrix_aligning_covariance_to_principal_axes(const Eigen::Matrix2d & covariance_matrix)`